src/core/dev/ring_simple.cpp

```diff
@@ -474,9 +474,14 @@ int ring_simple::poll_and_process_element_rx(uint64_t *p_cq_poll_sn,
 int ring_simple::poll_and_process_element_tx(uint64_t *p_cq_poll_sn)
 {
     int ret = 0;
-    RING_TRY_LOCK_RUN_AND_UPDATE_RET(m_lock_ring_tx,
-                                     m_p_cq_mgr_tx->poll_and_process_element_tx(p_cq_poll_sn));
-    return ret;
+    if (safe_mce_sys().doca_tx) {
+        RING_TRY_LOCK_RUN_AND_UPDATE_RET(m_lock_ring_tx, m_hqtx->poll_and_process_doca_tx());
```

Comment:
I changed to lock instead of try_lock for RX. There are reasons not to do try_lock

Reply:
Done.